### PR TITLE
Integrate golangci-lint, impi and shfmt

### PR DESCRIPTION
**Pull request by ferhatelmas**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request
**golangi-lint**: most comprehensive metalinter, that
does errcheck, static analysis, etc.

**impi**: ensures import ordering

**shfmt**: automatic shell script formatting similar to gofmt

Add linting to travis.

fixes: #12

### CODE REVIEW

**Comment by ferhatelmas** [comment]
This PR doesn't resolve #12 yet because there are still many issues that need to be fixed so that another PR can have a clean code but I wanted to show if the changes are inline with what we want.

**Comment by pascaldekloe** [comment]
Looks fine apart for the differ removals. Why did you do that? Now tests leave garbage on assertion failure, no?

**Comment by ferhatelmas** [comment]
@pascaldekloe `defer` could fail and they weren't checked but now they are checked (detected by `errcheck`). Otherwise, there shouldn't be any changes regarding messages.

**Comment by pascaldekloe** [comment]
Sure, that error check is fine. But you shouldn't eliminate the defers; create inner functions for that.

**Comment by ferhatelmas** [comment]
> But you shouldn't eliminate the defers; create inner functions for that.

Yes, that could be another option for implementation. Since these are tests, it seemed to me that moving these calls to later seemed more explicit.

If you want the other way, I can update.

**Comment by ferhatelmas** [comment]
@pascaldekloe I will update this part as you said and will fix the rest of the issues, too. Later, today, I will update.

**Comment by ferhatelmas** [comment]
One question is that how about changing package name to `stream` or `streamchat` because underscore in a package name is discouraged.

Since there is no nested package, it should be safe to do comparatively.

**Comment by pascaldekloe** [comment]
@Bogdan-D, you are in charge here ☝️. I think @ferhatelmas is right here.

**Comment by ferhatelmas** [comment]
This should be ready by now. 

If we want package name change, I can update but for now, I added boilerplate comment to silence the issue.

**Comment by Bogdan-D** [comment]
the problem is it will be unexpected for existing clients if package name will change

File: hack/run-lint.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+set -euo pipefail
+
+[[ -n ${DEBUG:-} ]] && set -x
+
+gopath="$(go env GOPATH)"
+
+if ! [[ -x "$gopath/bin/golangci-lint" ]]; then
+	echo >&2 'Installing golangci-lint'
+	curl --silent --fail --location \
+		https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b "$gopath/bin" v1.18.0
```
Comment (Bogdan-D) [inline hack/run-lint.sh]: please use install path from official url

File: hack/run-lint.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	install_impi
+fi
+
+"$gopath/bin/impi" \
+	--local github.com/GetStream/stream-chat-go \
+	--scheme stdThirdPartyLocal \
+	--skip "stream_chat_easyjson.go" \
+	./...
+
+install_shfmt() {
```
Comment (Bogdan-D) [inline hack/run-lint.sh]: why do we install shfmt?

**Comment by Bogdan-D** [comment]
Merged https://github.com/GetStream/stream-chat-go/pull/19

**Pull request closed without merging**